proto/redpanda/api/console/v1alpha1/security.proto

```diff
@@ -0,0 +1,239 @@
+syntax = "proto3";
+
+package redpanda.api.console.v1alpha1;
+
+import "buf/validate/validate.proto";
+
+// Role defines a role in the system.
+message Role {
+  // The name of the role.
+  string name = 1 [
+    (buf.validate.field).required = true,
+    (buf.validate.field).string.min_len = 1,
+    (buf.validate.field).string.max_len = 128,
+    (buf.validate.field).string.pattern = "^[a-zA-Z0-9-_]+$"
+  ];
+}
+
+// ListRolesRequest is the request for ListRoles.
+message ListRolesRequest {
+  // Filter options.
+  message Filter {
+    // Filter results only roles named with the prefix.
+    string name_prefix = 1 [
+      (buf.validate.field).string.max_len = 128,
+      (buf.validate.field).string.pattern = "^([a-zA-Z0-9-_]*)$"
+    ];
+
+    // Filter results to only roles with names which contain the string.
+    string name_contains = 2 [
+      (buf.validate.field).string.max_len = 128,
+      (buf.validate.field).string.pattern = "^([a-zA-Z0-9-_]*)$"
+    ];
+
+    // Return only roles assigned to this principal.
+    optional string principal = 3 [(buf.validate.field).string.max_len = 128];
```

Comment:
Reviving the discussion regarding the `optional` keyword here. I think similar to how no optional keyword was used for `name_prefix` and `name_contains` we shouldn't need to apply it here either. 

It's not possible to have role with an empty principal, so for this list request we can assume that `principal = ""` means we don't want to apply a filter on the principal name.

We follow Google's [AIP-149](https://google.aip.dev/149)

Quote:
> Services defined in protocol buffers should use the optional keyword for primitives if and only if it is necessary to distinguish setting the field to its default value (0, false, or empty string) from not setting it at all

Reply:
I have removal the optional property from that field.